### DISCUSSION TITLE
New plugin type: transformations

### DISCUSSION THREAD

**Discussion by mckn** [grafana/grafana#38540] [Plugins]
This is a request given by @Willena in the [Roadmap issue](https://github.com/grafana/grafana/issues/36228#issuecomment-897077347)

Since the introduction of transformations, I've seen that the team keeps adding more transforms to Grafana (it is great !).
I would love to see a new plugin API dedicated to transformations. So that transformations that are not considered important for Grafana Core could still be added using a plugin.

Any thoughts on that ?

**Comment by mckn** [comment]
I think this is a good idea! I would suggest that it might be worth adding support for backend transformations as well.

**Comment by pkolyvas** [comment]
I also love this idea - it could open up some fascinating opportunities for the community to help us define what kind of  transformations they find most valuable.

**Comment by mikhail-vl** [comment]
There are multiple use cases when data from data sources should be transformed in a unique way not supported by existing plugins.
In this case, we use the Apache ECharts panel with JavaScript code to transform data.

As soon as transformation plugins were introduced, we would create a JavaScript Code Transformation.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!

**Comment by adminy** [comment]
even as a builtin please, any javascript arbitrary field or row transformation would move grafana miles into the future.
